cpp/include/cudf/detail/stream_compaction.hpp

```diff
@@ -82,10 +82,36 @@ std::unique_ptr<table> unique(
 std::unique_ptr<table> distinct(
   table_view const& input,
   std::vector<size_type> const& keys,
+  duplicate_keep_option keep          = duplicate_keep_option::KEEP_ANY,
   null_equality nulls_equal           = null_equality::EQUAL,
   rmm::cuda_stream_view stream        = rmm::cuda_stream_default,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Get the row indices that denote all distinct rows in `input`.
+ *
+ * Given an `input` table_view, each row index is copied to the output array in an
+ * unspecified order if the corresponding row of `keys` columns is distinct (no other
+ * equivalent row exists in the table).
+ *
+ * Note that the returning gather map is compact: all the indices are valid.
```

Comment:
This statement is confusing. Why wouldn't all of the indices be valid?

Reply:
I think this is a leftover when addressing review comments.

@ttnghia The idea is to avoid the word "gather map" in the doc here and use "row indices" consistently.

Reply:
This means there is no out-of-bound indices. But the statement may not be necessary. I remove it.